src/session-pool.ts

```diff
@@ -744,9 +774,14 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
         });
 
         amount -= sessions.length;
+        nReturned += sessions.length;
       } catch (e) {
         this._pending -= amount;
         this.emit('createError', e);
+        span.addEvent(
```

Comment:
We can remove this span as we are setting exception below.

Reply:
I think we should keep it despite the exception because if for example it gets caught, that span should still have recorded the status at that terminal point.